arm/Microsoft.ServiceFabric/clusters/deploy.bicep

```diff
@@ -0,0 +1,398 @@
+    supportLogStorageAccountKeys: supportLogStorageAccount.listKeys()
+    location: location
+    tags: tags
+  }
+}
+
+// Service Fabric cluster resource lock
+resource serviceFabricCluster_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lock != 'NotSpecified') {
+  name: '${serviceFabricCluster.name}-${lock}-lock'
+  properties: {
+    level: lock
+    notes: (lock == 'CanNotDelete') ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+  }
+  scope: serviceFabricCluster
+}
+
+// Service Fabric cluster RBAC assignment
+module serviceFabricCluster_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${serviceFabricCluster.name}-${uniqueString(deployment().name, location)}-Rbac-${index}'
+  params: {
+    roleAssignmentObj: roleAssignment
+    builtInRoleNames: builtInRoleNames
+    resourceName: serviceFabricCluster.name
+  }
+}]
+
+// Service Fabric cluster application types
+// module serviceFabricCluster_applicationTypes '.bicep/nested_applicationTypes.bicep' = [for applicationType in serviceFabricApplicationTypes: {
+//   name: '${serviceFabricCluster.name}-${applicationType.name}'
+//   params: {
+//     applicationTypeObj: applicationType
+//     clusterName: serviceFabricCluster.name
+//     location: location
+//     tags: tags
+//     properties: !empty(applicationType.properties) ? applicationType.properties : null
+//   }
+// }]
+
+// // Service Fabric cluster applications
+// module serviceFabricCluster_applications '.bicep/nested_applications.bicep' = [for application in serviceFabricClusterApplications: {
+//   name: '${serviceFabricCluster.name}-${application.name}'
+//   params: {
+//     applicationObj: application
+//     clusterName: serviceFabricCluster.name
+//     location: location
+//     tags: tags
+//     identity: !empty(application.identity) ? application.identity : null
+//     properties: {
+//       managedIdentities: !empty(application.managedIdentities) ? application.managedIdentities : null
+//       maximumNodes: contains(application, 'maximumNodes') ? application.maximumNodes : 1
+//       metrics: !empty(application.metrics) ? application.metrics : null
+//       minimumNodes: contains(application, 'minimumNodes') ? application.minimumNodes : 0
+//       parameters: !empty(application.parameters) ? application.parameters : null
+//       removeApplicationCapacity: contains(application, 'removeApplicationCapacity') ? application.removeApplicationCapacity : false
+//       typeName: !empty(application.typeName) ? application.typeName : null
+//       typeVersion: !empty(application.typeVersion) ? application.typeVersion : null
+//       upgradePolicy: !empty(application.upgradePolicy) ? application.upgradePolicy : null
+//     }
+//   }
+//   dependsOn: [
+//     serviceFabricCluster_applicationTypes
+//   ]
+// }]
+
+// Outputs section
+output serviceFabricCluster object = serviceFabricCluster
```

Comment:
Needs output descriptions

Reply:
Added.

Reply:
Not sure if we want to return the entire object. Let's remove that